public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx

```diff
@@ -0,0 +1,278 @@
+import { ActionType } from "../../utils/constants";
+import { DarkModeConsumer } from "../../../../components/DarkMode";
+import EuiFormCustomLabel from "../EuiFormCustomLabel";
+
+const radios = [
+  {
+    id: "no",
+    label: "No",
+  },
+  {
+    id: "yes",
+    label: "Yes",
+  },
+];
+
+export default class ShrinkUIAction implements UIAction<ShrinkAction> {
+  id: string;
+  action: ShrinkAction;
+  type = ActionType.Shrink;
+
+  constructor(action: ShrinkAction, id: string = makeId(), useJsons: boolean = false) {
+    this.action = { ...action };
+    this.id = id;
+    // When the jsons are edited by the user, don't overwrite the json. When cloned elsewhere, take the actual object as source of truth
+    if (!useJsons) {
+      this.action.target_index_name_template_json = JSON.stringify(this.action.shrink.target_index_name_template, null, 4);
+      this.action.aliases_json = JSON.stringify(this.action.shrink.aliases, null, 4);
+      this.action.force_unsafe_input = this.action.shrink.force_unsafe ? "yes" : "no";
+    }
+  }
+
+  content = () => `Shrink`;
+
+  clone = (action: ShrinkAction = this.action) => new ShrinkUIAction(action, this.id);
+
+  cloneUsingString = (action: ShrinkAction) => new ShrinkUIAction(action, this.id, true);
+
+  isValid = () => {
+    return this.isValidNumShards() && this.isValidAliasesJson() && this.isValidIndexNameTemplateJson();
+  };
+
+  isValidNumShards = () => {
+    const shrink = this.action.shrink;
+    const numSet = [shrink.num_new_shards != null, shrink.percentage_of_source_shards != null, shrink.max_shard_size != null].filter(
+      (it) => it
+    ).length;
+    return numSet == 1;
+  };
+
+  isValidAliasesJson = () => {
+    if (this.action.aliases_json && this.action.aliases_json?.length > 0) {
+      try {
+        JSON.parse(this.getAliasesJsonString(this.action) as string);
+        return true;
+      } catch (err) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  getAliasesJsonString = (action: ShrinkAction) => {
+    const shrink = action.shrink;
+    return action.hasOwnProperty("aliases_json") ? action.aliases_json : JSON.stringify(shrink.aliases, null, 4);
+  };
+
+  isValidIndexNameTemplateJson = () => {
+    if (this.action.target_index_name_template_json && this.action.target_index_name_template_json?.length > 0) {
+      try {
+        JSON.parse(this.getIndexNameTemplateJsonString(this.action) as string);
+        return true;
+      } catch (err) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  getIndexNameTemplateJsonString = (action: ShrinkAction) => {
+    if (action.hasOwnProperty("target_index_name_template_json")) {
+      return action.target_index_name_template_json;
+    } else {
+      return JSON.stringify(action.shrink.target_index_name_template, null, 4);
+    }
+  };
+
+  render = (action: UIAction<ShrinkAction>, onChangeAction: (action: UIAction<ShrinkAction>) => void) => {
```

Comment:
Not sure, how we are doing for other actions but may be differentiating optional and required fields by an `*` or calling it out in words might be helpful

Reply:
Added and demoed in a new screenshot